src/RelayClient.ts

```diff
@@ -44,16 +76,206 @@ class RelayClient extends EnvelopingEventEmitter {
+    return Promise.resolve(constants.Zero);
+  };
+
+  public estimateInternalCallGas = async (
+    _params: EstimateInternalGasParams
+  ): Promise<BigNumber> => {
+    return Promise.resolve(constants.Zero);
   };
 
-  // TODO: lift the ts exception once method is used
   // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-  // @ts-ignore
+  // @ts-ignore FIXME: remove the ts ignore once the method is consumed
+  private _getEnvelopingRequestDetails = async (
+    envelopingRequest: UserDefinedEnvelopingRequest,
+    {
+      isSmartWalletDeploy,
+      preDeploySWAddress,
+      forceGasLimit,
+      forceGasPrice,
+      internalEstimationCorrection,
+      addCorrection,
+    }: RequestConfig
+  ): Promise<EnvelopingRequest> => {
+    const isDeployment: boolean = isDeployRequest(
+      envelopingRequest as EnvelopingRequest
+    );
+    const {
+      relayData: { callForwarder },
+      request: { data, from, to, tokenContract },
+    } = envelopingRequest;
+
+    if (!callForwarder) {
+      throw new Error(MISSING_CALL_FORWARDER);
+    }
+
+    const callVerifier: PromiseOrValue<string> =
+      envelopingRequest.relayData.callVerifier ||
+      this._envelopingConfig[
+        isDeployment ? 'deployVerifierAddress' : 'relayVerifierAddress'
+      ];
+
+    if (!callVerifier) {
+      throw new Error('No call verifier present. Check your configuration.');
+    }
+
+    const relayServerUrl = this._envelopingConfig.preferredRelays.at(0)?.url;
+
+    if (!relayServerUrl) {
+      throw new Error(
+        'Check that your configuration contains at least one preferred relay with url.'
+      );
+    }
+    const feesReceiver: PromiseOrValue<string> = (
+      await new HttpClient().getChainInfo(relayServerUrl)
+    ).feesReceiver;
+
+    if (!feesReceiver) {
+      throw new Error('Hub info does not contain fees receiver');
+    }
+
+    const gasPrice: PromiseOrValue<BigNumberish> =
+      forceGasPrice ??
+      (envelopingRequest.relayData.gasPrice ||
+        (await this._calculateGasPrice()));
+
+    if (!gasPrice || BigNumber.from(gasPrice).isZero()) {
+      throw new Error('Could not get gas price for request');
+    }
+
+    if (!data) {
+      throw new Error('Field `data` is not defined in request body.');
+    }
+
+    if (!from) {
+      throw new Error('Field `from` is not defined in request body.');
+    }
+
+    if (!to) {
+      throw new Error('Field `to` is not defined in request body.');
+    }
+
+    const value = envelopingRequest.request.value ?? constants.Zero;
+    const tokenAmount = envelopingRequest.request.tokenAmount ?? constants.Zero;
+
+    if (!tokenContract) {
+      throw new Error('Field `tokenContract` is not defined in request body.');
+    }
+
+    const nonce =
+      envelopingRequest.request.nonce ||
+      (await IForwarder__factory.connect(
+        callForwarder.toString(),
+        this._provider
+      ).nonce());
+
+    if (!nonce || BigNumber.from(nonce).isZero()) {
```

Comment:
nonce can be zero if its the first transaction, so the nonZero validation can be removed

Reply:
ah yea I forgot... tx